dGame/dMission/Mission.cpp

```diff
@@ -481,8 +481,10 @@ void Mission::YieldRewards() {
         inventoryComponent->AddItem(pair.first, count);
     }
 
-    if (info->reward_currency > 0) {
-        character->SetCoins(character->GetCoins() + info->reward_currency, info->isMission);
+    if (info->reward_currency > 0 && info->isMission) {
```

Comment:
You could just do a nested if clause and first check currency, then isMission, and assign the loot source type accordingly and only have one SetCoins call. Your way works too, but is a bit unusual.

Reply:
I see what you mean.  I [re-did my statements](https://github.com/DarkflameUniverse/DarkflameServer/pull/291/commits/e84811bdcee9218b4c88c5ad6a3a4e63298cebcf) there to have only one SetCoins statement.